### PR TITLE
#306 dvc-file-format updated

### LINKED ISSUE #306: describe "meta" field in stage file and that we are preserving comments
https://github.com/iterative/dvc/issues/1799

### PR DESCRIPTION
**Pull request by ryokugyu**
Fixes #306 
meta entry updated and also mentioned that comments are persistent now.

### CODE REVIEW

File: static/docs/user-guide/dvc-file-format.md [hunk 1]
```diff
@@ -60,3 +64,10 @@ A metric entry consists of such fields:
 - `type`: type of the metrics file (e.g. raw/json/tsv/htsv/csv/hcsv);
 - `xpath`: path within the metrics file to the metrics data(e.g. `AUC.value` for
   `{"AUC": {"value": 0.624321}}`);
+
+A meta entry consists of `key:value` pairs such as `name: john`. A meta entry
+can have any structure and contain any number of attributes. `"meta: string"` is
+also possible, it doesn't necessarily need to contain dictionary always.
+
+Comments can be added to the .dvc file using `# comment` syntax. Comments are
+preserved between multiple execution of  `dvc repro <filename.dvc>` command.
```
Comment (shcheklein) [inline static/docs/user-guide/dvc-file-format.md]: of the `dvc repro` command, no need for <filename.dvc> - it's optional.

probably we need to use `multiple executions`

need to clarify, if meta and comments are preserved when we do `dvc run` that overwrites the file again - please, clarify it on #dev-general

Comment (shcheklein) [reply static/docs/user-guide/dvc-file-format.md]: the same for `dvc add` and `dvc import` - they also create DVC-files. We need to clarify here, on in their documentation what happens to comments/meta if we run them again.

also, `dvc commit` updates DVC-files - let's clarify if it also preserves the comments.

Comment (ryokugyu) [reply static/docs/user-guide/dvc-file-format.md]: > need to clarify, if meta and comments are preserved when we do `dvc run` that overwrites the file again - please, clarify it on #dev-general

meta and comments are not preserved between `dvc run` command.

File: static/docs/user-guide/dvc-file-format.md [hunk 2]
```diff
@@ -30,6 +30,10 @@ outs:
       xpath: AUC
     path: metrics.json
 locked: True
+# Comments are also persisted between dvc repro command.
```
Comment (shcheklein) [inline static/docs/user-guide/dvc-file-format.md]: add dvc commit

Comment (shcheklein) [reply static/docs/user-guide/dvc-file-format.md]: also let's add a second comment that dvc run/add/import overwrite the file

File: static/docs/user-guide/dvc-file-format.md [hunk 3]
```diff
@@ -60,3 +64,11 @@ A metric entry consists of such fields:
 - `type`: type of the metrics file (e.g. raw/json/tsv/htsv/csv/hcsv);
 - `xpath`: path within the metrics file to the metrics data(e.g. `AUC.value` for
   `{"AUC": {"value": 0.624321}}`);
+
+A meta entry consists of `key:value` pairs such as `name: john`. A meta entry
+can have any structure and contain any number of attributes. `"meta: string"` is
+also possible, it doesn't necessarily need to contain dictionary always.
+
+Comments can be added to the .dvc file using `# comment` syntax. Comments
```
Comment (shcheklein) [inline static/docs/user-guide/dvc-file-format.md]: same here, let's mention that dvc run/add/import when they overwrite/update the file information is lost

File: static/docs/commands-reference/import.md [hunk 4]
```diff
@@ -195,6 +195,8 @@ The `etag` field in the DVC file contains the ETag recorded from the HTTP
 request. If the remote file changes, the ETag changes, letting DVC know when the
 file has changed.
 
+While executing `dvc import` command, if user overwrites the `.dvc` file,
+comments  and meta values are not preserved between multiple executions.
```
Comment (shcheklein) [inline static/docs/commands-reference/import.md]: remove extra space

File: static/docs/user-guide/dvc-file-format.md [hunk 5]
```diff
@@ -30,6 +30,12 @@ outs:
       xpath: AUC
     path: metrics.json
 locked: True
+
+#Comments are persisted between multiple executions of dvc repro/commit.
```
Comment (shcheklein) [inline static/docs/user-guide/dvc-file-format.md]: Add space between # and the first word here and below

File: static/docs/user-guide/dvc-file-format.md [hunk 6]
```diff
@@ -60,3 +66,15 @@ A metric entry consists of such fields:
...
   `{"AUC": {"value": 0.624321}}`);
+
+A meta entry consists of `key:value` pairs such as `name: john`. A meta entry
+can have any structure and contain any number of attributes. `"meta: string"` is
+also possible, it doesn't necessarily need to contain dictionary always.
+
+Comments can be added to the .dvc file using `# comment` syntax.
+
+Comments and meta values are preserved between multiple executions of `dvc repro`
+ and `dvc commit` commands.
```
Comment (shcheklein) [inline static/docs/user-guide/dvc-file-format.md]: remove extra space. Merge this paragraph with the previous one, please.

**Pull request merged**